util/annotations/annotations.go

```diff
@@ -150,45 +160,83 @@ var (
 	HistogramIgnoredInAggregationInfo       = fmt.Errorf("%w: ignored histogram in", PromQLInfo)
 )
 
-type annoErr struct {
+type annoErr interface {
+	error
+	// We can define custom merge functions to merge annoErrs with the same raw error string.
+	merge(annoErr) annoErr
```

Comment:
I think I understand now what you are up to here, but having two different levels of merging is complex enough so that we need to explain this better here. So there is the exported `Merge` to merge two instances of `Annotations` with each other, but then there is this unexported `merge` to merge one individual annotation with another one.